numba/tests/test_recarray_usecases.py

```diff
@@ -144,5 +146,48 @@ def test_usecase5(self):
         self._test_usecase2to5(usecase5, self.aligned_dtype)
 
 
+class TestRecArrayCreation(TestCase):
+    def test_creation(self):
+        a_b_type = np.dtype([('a', 'i8'), ('b', 'f8')], align=True)
+        x = np.array([(1, 2.), (3, 4.)], a_b_type)
+
+        @njit
+        def create():
+            new = np.array([(1., 2.), (3., 4.)], a_b_type)
+            return new
+
+        with self.subTest("creation from list of tuple"):
+            self.assertTrue(np.array_equal(create(), x))
+
+        @njit
+        def create():
+            new = np.array([[1., 2.], [3., 4.]], a_b_type)
+            return new
+
+        with self.subTest("creation from nested list"):
+            self.assertTrue(np.array_equal(create(), x))
```

Comment:
These tests might be better as:
```suggestion
            self.assertTrue(np.array_equal(create(), jit(create)()))
```
after removing the `@jit` decorator above.